src/Microsoft.Azure.SignalR/ServiceOptionsExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal static class ServiceOptionsExtensions
+    {
+        public static void Validate(this ServiceOptions options)
+        {
+            if (options.DisconnectTimeoutInSeconds.HasValue &&
+                (options.DisconnectTimeoutInSeconds < 1 || options.DisconnectTimeoutInSeconds > 300))
+            {
+                throw new ArgumentOutOfRangeException("DisconnectTimeoutInSeconds", "Value should be [1,300].");
```

Comment:
* ArgumentException, the argument is 'options'
* Can we check it when set the property `DisconnectTimeoutInSeconds`?

Reply:
Not able to move this into ServiceOptionsSetup cause the value is not coming from file configuration but code option settings.

AspNetCore support [`IOptionsValidate`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1) after 2.2 which is not compatible with 2.1, so choose to use `PostConfigure` to do the validation.

Reply:
I think `ArgumentOutOfRangeException` should be more appropriate for our scenario. https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception?view=netcore-3.1#remarks